src/components/Courses/Courses.jsx

```diff
@@ -1,45 +1,113 @@
-import React, { useState, useMemo } from 'react';
-import './Courses.css';
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 import PropTypes from 'prop-types';
 import SearchBar from './components/SearchBar/SearchBar';
 import CourseCard from './components/CourseCard/CourseCard';
 import Button from '../../common/Button/Button';
+import CourseInfo from '../CourseInfo/CourseInfo';
+import { api_endpoints, storageKeys, paths } from '../../constants';
+import './Courses.css';
 
-const Courses = ({ courses, authors, onAddCourseClick }) => {
+const Courses = ({ onAddCourseClick }) => {
```

Comment:
From what I can see - you already fetch for courses and authors either in `App.jsx` and in this `Courses.jsx` component

As I understand, this component will contain all the necessary info and functionality, so I'd suggest removing the fetching in `App.jsx` and only having it here.